### PR TITLE
Image Accumulation + User Camera

### PR DESCRIPTION
**Pull request by Jebbly**
Renders to an offscreen image to accumulate samples. The path tracing is not complete yet and the sampling is just a proof-of-concept for the image accumulation.

Also includes a camera that will only reset the accumulation after moving.

### CODE REVIEW

File: drivers/benchmark_graphics.cpp [hunk 1]
```diff
@@ -27,7 +28,11 @@ int main(int argc, char *argv[]) {
     }
     auto scene = build_scene(context, std::move(objects));
 
-    while (!should_exit(context)) {
-        render_frame(context, scene);
+    while (!window->should_close()) {
+        window->poll_events();
+        camera->handle_keys(0.0001f);
```
Comment (RArbore) [inline drivers/benchmark_graphics.cpp]: Where is camera supposed to be defined?

File: shaders/Raw_Color_rchit.glsl [hunk 2]
```diff
@@ -23,6 +23,10 @@ void main() {
     vec3 world_obj_pos = gl_ObjectToWorldEXT * vec4(0.0, 0.0, 0.0, 1.0);
 
     vec3 voxel_sample_pos = gl_WorldToObjectEXT * vec4(world_ray_pos, 1.0);
-    ivec3 volume_load_pos = ivec3(voxel_sample_pos - 0.5 * voxel_normals[gl_HitKindEXT]);
-    hit = imageLoad(volumes[volume_id], volume_load_pos);
+    ivec3 volume_load_pos = ivec3(voxel_sample_pos * vec3(imageSize(volumes[volume_id])) - 0.5 * voxel_normals[gl_HitKindEXT]);
+
+    payload.hit = true;
+    payload.world_position = voxel_sample_pos;
+    payload.world_normal = voxel_normals[gl_HitKindEXT];
```
Comment (RArbore) [inline shaders/Raw_Color_rchit.glsl]: voxel_normals contains object normals, not world normals.

**Review by RArbore — CHANGES_REQUESTED** [review]
Looks good, but there's a couple of things to fix for build. Also, I tried defining camera myself, and the screen was just black.

File: drivers/benchmark_graphics.cpp (hunk 1 above)
Comment (Jebbly) [reply drivers/benchmark_graphics.cpp]: Camera should be created separately, after creating the context. I just pushed an example to ``ivs.cpp``.

**Pull request merged**